demo/demo_secondary.py

```diff
@@ -78,8 +80,10 @@ def clean_slate(
   global _primary_host
   global _primary_port
   global nonce
-  global client_directory
+  global CLIENT_DIRECTORY
   global attacks_detected
+  atexit.register(clean_up) # To delete the temp pinned file and folder
```

Comment:
As noted for the other module, this shouldn't happen in module-level code that would be executed on import.